src/mem/sizeclasstable.h

```diff
@@ -179,7 +179,7 @@ namespace snmalloc
 
     auto rsize = sizeclass_to_size(sc);
 
-    if constexpr (bits::is64())
+    if constexpr (bits::BITS >= 64)
```

Comment:
I think we have two concepts here that feel slightly conflated:

 - What is the size of a C/C++ integer type that can express any address?  Or possibly what is the type that can represent any address.
 -  How many of those bits are actually ever used?

This wants to be gated on something like `std::is_same_v<uint64_t, Pal::address_t>`.

Reply:
Hm.  I think this might want to be gated rather on `sizeof(offset) >= 8`, as that will more closely determine whether the expression overflows?  @mjp41?

Reply:
It is really that the code is designed to work with the existing potential offsets with 64bit multiplication.  If we had a larger multiply, then that would be fine as long as we didn't have larger offsets.

This code still needs some tidying from the snmalloc2 refactor. So I will harden the choices when I do that.   So this is fine for now.